### ISSUE TITLE
Dynamic stacked/clustered column chart

### ISSUE THREAD

**Issue by aryesgrace** [microsoft/charticulator#473]
**TLDR;** how can I apply [this solution](https://github.com/microsoft/charticulator/issues/219)  to apply to my data? Specifically, how can the faux-stacked effect be achieved when each shape doesn't start on the axis at zero? 

<img width="700" alt="NadiaFankTemplate" src="https://user-images.githubusercontent.com/81455088/112711194-3ec1be80-8f12-11eb-9b2f-3a49826a895b.png">

My data (graph would also need an extra 'column', but I assume that is as easy as adding a third plot segment)
<img width="700" alt="ExampleWData" src="https://user-images.githubusercontent.com/81455088/112711758-96622900-8f16-11eb-9100-731974e1a0eb.png">

**Background**

I'm usually pretty good at picking up a new tool (self-taught Power Automate, Power Query, VBA, JavaScript, Adobe Suite etc). But Charticulator is really challenging me, and (like everyone) I'm struggling to find training material. 

I came to Charticulator because I wanted to achieve something similar to the stacked and clustered column chart for Power BI on the thread below. 
[https://github.com/microsoft/charticulator/issues/219](https://github.com/microsoft/charticulator/issues/219)

The solution provided by @nadiafank was great because it demonstrated what was possible, and had been set up with a wide-dataset (per my data). Unfortunately I was unable to duplicate the layout because I couldn't determine how the faux-stacked shapes were achieved per my initial question above.

The first (lowest) shape was either data driven height value on the shape or anchored to the data-driven Y axis, and started at 0. Easy enough. The subsequent (higher) shapes in each column looked like they had a data driven height value on the shape, but how did it know where to start on the Y axis? I tried anchoring to the top of the shape beneath, but that didn't seem to work. (At least, the scale/values were definitely all wrong)

After a few hours tinkering I converted my data to a narrow format (thinking that I'd at least be able to experiment with the 'stack Y' layout) and included two levels of categories. 

![ExampleData](https://user-images.githubusercontent.com/81455088/112709588-965a2d00-8f06-11eb-8589-b37260a47178.png)
_(Sanitized)_

I wanted three 'stack X' columns to match the 'Categories', and the 'SubCategories' to appear as 'stack Y' on their applicable columns. On this occasion only one of my categories had subtypes, so I tried setting it up with three separate segments each filtered by a different 'Category', then added a nested chart in my 'spend' segment overlayed on my existing shape.

It has come close to what I need, but I had a lot of trouble getting my axes to match, the width of my nested table isn't dynamic, and I will need to manually set my legend colors to match the sub categories since I can't get them to be dynamic (due to separate shape/segments and nested table).
![Example](https://user-images.githubusercontent.com/81455088/112709501-f4d2db80-8f05-11eb-8d10-9104b93e0ac0.png)
_(Sanitized)_

[File download](https://drive.google.com/file/d/14iz0MrHBs54zuZ1MJ6rBkNucFuVFVi6C/view?usp=sharing)

Point is, I'm sure I've done it horribly wrong and it won't hold up in Power BI. What would be AMAZING is being able to create a fully dynamic template using narrow data & generic headings, so that next time I can just reuse the visual and have all my jobs/categories/subcategories be completely dynamic. Not to mention it would be any almost plug-and-play solution for other users. Would there be a way to achieve this?

A few of the other issues I've been having:
- How to force multiple segments/glyphs/shapes to use the same (or duplicated) axes, especially when using different filters per segment.
- How to force my Y axis maximum to be slightly higher than my maximum values (for aesthetics), while still maintaining dynamic values (for stability).
- How and when to anchor shapes to guides. Guides are referenced constantly in the training material. I get that they somehow anchor the elements but I don't understand what they themselves lock to (is there an underlying coordinate system or true zero?), or whether they can/should apply across multiple segments/glyphs.
- Why my glyphs on the 2nd and 3rd segments (which should be on the same y axis as the 1st segment) are floating above/below the X axis at a seemingly random distance when I apply the same axis to multiple segments? (I'd understand if it was just the scale that changed due to the filters on my segments, but they seem to expand into negative values or float halfway up the graph)

![Example_Floaties](https://user-images.githubusercontent.com/81455088/112710217-28643480-8f0b-11eb-89cf-dc185485fb01.jpg)

Thanks very much for your time. Really appreciate any advice anyone can give me!

**Comment by bongshin** [comment]
@aryesgrace --- I created this custom bar chart using your data. 
![image](https://user-images.githubusercontent.com/9288805/112780164-1bc00900-8ffd-11eb-8f56-63451c515d21.png)
Here is the chart file and the csv file created with your data: [Custom Bar Chart.zip](https://github.com/microsoft/charticulator/files/6218889/Custom.Bar.Chart.zip). In this case, it makes sense to format the data in wide form. Note that I added a column (Spend), which sums up the five types of spend to use the value in the label, and created a color legend using data column names.

You can make all the rectangles to use the same scale by putting them under the same scale (see Scale 1 in the Scales panel below) using the drag-and-drop interaction. 

![image](https://user-images.githubusercontent.com/9288805/112780120-00ed9480-8ffd-11eb-91b2-2804961f5857.png)

Top, Bottom, Left, and Right default guides are placed relative to the chart canvas using its Margin attribute values. Middle and Center guides are in the middle of Top & Bottom and Left & Right, respectively. User-defined guides can be placed relative to one of the four default guides.

---
If you want to see Charticulator further developed and eventually integrated into Power BI Desktop, please add your vote to this idea, https://ideas.powerbi.com/ideas/search-ideas/?q=charticulator.

**Comment by aryesgrace** [comment]
@bongshin, thank you so much for your help! This is amazing. I hate not knowing how to do something myself. 

There were a few changes to the Y axis I was hoping you could provide advice on if possible?
- how can I add horizontal grid lines? (I know it is possible on plot segment axis, but cannot find on the legend axis)
- how can I increase the scale/height of the chart? (I want to give a little white space above the 8.93M maximum value, say to 10M)
- Increase the number of horizontal / y axis tick marks / gridlines to be every 500k

The Y axis in your version appears to be a legend which has limited editing options, and when I add an axis to the plot segment the following happens. Can you explain what is happening here? (i.e. the white space below each glyph?)

![Example3](https://user-images.githubusercontent.com/81455088/112904796-d4389a80-912c-11eb-8c25-871773128e3b.png)

Thank you!!!
